File: Tasks/day.js
// Refactor following solution
// Get day number

'use strict';

const DAYS = ['sun', 'mon', 'tue', 'wed', 'thu', 'fri', 'sat'];

const parseDay = (day) => {
  const index = DAYS.findIndex((d) => day.startsWith(d));
  return index + 1 || -1;
};

module.exports = parseDay;

File: Tasks/Day.java
import java.util.List;

public class Day {
    private static final List<String> DAYS = List.of("sun", "mon", "tue", "wed", "thu", "fri", "sat");

    public static int parseDay(String input) {
        if (input == null || input.isEmpty()) {
            return -1;
        }

        for (String day : DAYS) {
            if (input.startsWith(day)) {
                return DAYS.indexOf(day) + 1;
            }
        }

        return -1;
    }
}
